contributing_to_docs/doc_guidelines.adoc

```diff
@@ -1040,70 +1041,123 @@ This renders as:
 > docker-registry     <none>                                    name=registrypod    172.30.17.158       5001
 > ----
 
-The following guidelines go into more detail about specific requirements and
-recommendations when using code blocks:
+* Any `[source]` metadata must go on the line directly before the code block. Also, do not insert a space in between the `[source]` tag and the metadata.
+//Context: https://github.com/openshift/openshift-docs/pull/64373
++
+For example:
++
+....
+
+[source,terminal]
+----
+$ oc get se
+----
 
-* If a step in a procedure is to run a command, make sure that the step
-text includes an explicit instruction to "run" or "enter" the command. In most cases,
-use one of the following patterns to introduce the code block:
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1
+kind: Scheduler
+metadata:
+  name: cluster
+# ...
+spec:
+  defaultNodeSelector: node-role.kubernetes.io/app=
+# ...
+----
+....
+
+* For Bash "here" documents use `[source,terminal]`, such as the following example:
++
+....
+[source,terminal]
+----
+$ cat <<EOF| oc create -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: mlistener
+  labels:
+    app: multicast-verify
+EOF
+----
+....
++
+[NOTE]
+====
+For bash scripts, use `[source,bash]`. For example:
+[source,bash]
+----
+#!/bin/bash
+
+# optional argument handling
+if [[ "$1" == "config" ]]
+then
+    echo $KUBECONFIG
+    exit 0
+fi
+
+echo "I am a plugin named kubectl-foo"
+----
+====
+
+The following guidelines go into more detail about specific requirements and recommendations when using code blocks:
+
+[[step-with-command]]
+==== Procedure step that introduces a command
+If a step in a procedure is to run a command, make sure that the step text includes an explicit instruction to "run" or "enter" the command. In most cases, use one of the following patterns to introduce the code block:
 
 ** <Step description> by running the following command:
 ** <Step description> by entering the following command:
 ** <Step description>, run the following command:
 ** <Step description>, enter the following command:
 
-* Do NOT use any markup in code blocks; code blocks generally do not accept any markup.
+[[no-markup-codeblock]]
+==== No markup in code blocks
+Do NOT use any markup in code blocks; code blocks generally do not accept any markup.
+
+[[empty-line-before-codeblock]]
+==== Empty line before code blocks
+For all code blocks, you must include an empty line above a code block (unless that line is introducing block metadata, such as `[source,terminal]` for syntax highlighting).
 
-* For all code blocks, you must include an empty line above a code block (unless
-that line is introducing block metadata, such as `[source,terminal]` for syntax
-highlighting).
-+
 Acceptable:
-+
+
```

Comment:
Should we put a `[source,terminal]` (or whatever) before the commands in these examples for consistency?

Reply:
Sorry, but adding it to this example would render the guideline moot.
An empty line before a code block is required only when it is **not** preceded by a block metadata such as` [source,terminal]`, which is also stated in the guideline.